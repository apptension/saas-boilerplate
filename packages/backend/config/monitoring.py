import sentry_sdk
from sentry_sdk.integrations.django import DjangoIntegration
from sentry_sdk.integrations.logging import ignore_logger
from sentry_sdk.scope import add_global_event_processor
from rest_framework_simplejwt.exceptions import InvalidToken

ignore_logger('graphql.execution.utils')


@add_global_event_processor
def processor(event, hint):
    if event.get("type") == "transaction" and event.get("transaction") == "/lbcheck":
        return None
    return event


def init(dsn, environment_name, traces_sample_rate):
    sentry_sdk.init(
        dsn=dsn,
        integrations=[DjangoIntegration()],
        traces_sample_rate=traces_sample_rate,
        send_default_pii=True,
        environment=environment_name,
        ignore_errors=[InvalidToken],
    )
