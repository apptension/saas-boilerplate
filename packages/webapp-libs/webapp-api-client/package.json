{
  "name": "@sb/webapp-api-client",
  "version": "4.1.1",
  "type": "commonjs",
  "scripts": {
    "test": "jest",
    "test:watch": "jest --watch",
    "lint": "eslint .",
    "graphql-codegen": "graphql-codegen",
    "type-check": "tsc --noEmit --project tsconfig.lib.json"
  },
  "dependencies": {
    "@sb/webapp-core": "workspace:*",
    "apollo-upload-client": "^17.0.0",
    "axios": "^1.8.2",
    "axios-case-converter": "^0.9.0",
    "http-status-codes": "^2.3.0"
  },
  "devDependencies": {
    "@graphql-codegen/add": "^5.0.0",
    "@graphql-codegen/client-preset": "^4.1.0",
    "@graphql-codegen/import-types-preset": "^2.2.6",
    "@graphql-codegen/typescript": "^4.0.1",
    "@graphql-codegen/typescript-graphql-request": "^5.0.0",
    "@graphql-codegen/typescript-operations": "^4.0.1",
    "@graphql-tools/merge": "^9.0.1",
    "@types/apollo-upload-client": "^17.0.5",
    "fs-extra": "^11.2.0",
    "graphql-ws": "^5.14.2",
    "msw": "^1.3.2",
    "subscriptions-transport-ws": "^0.11.0"
  }
}
