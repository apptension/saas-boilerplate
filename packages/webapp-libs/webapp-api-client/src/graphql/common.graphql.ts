import { gql } from '.';

export const pageCursorsFragment = gql(/* GraphQL */ `
  fragment pageCursorsFragment on PageCursors {
    around {
      cursor
      isCurrent
      page
    }
    first {
      cursor
      isCurrent
      page
    }
    last {
      cursor
      isCurrent
      page
    }
    next {
      cursor
      isCurrent
      page
    }
    previous {
      cursor
      isCurrent
      page
    }
  }
`);
