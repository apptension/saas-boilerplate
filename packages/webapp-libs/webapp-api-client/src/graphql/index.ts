import { client as apolloClient, emitter as apolloEmitter, invalidateApolloStore } from './apolloClient';

export * from './types';
export * from './__generated/gql';
export * from './__generated/gql/graphql';
export * from './common.graphql';

export { apolloClient, apolloEmitter, invalidateApolloStore };
