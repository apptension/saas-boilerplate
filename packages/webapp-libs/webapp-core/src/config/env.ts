// @ts-nocheck

export const ENV = {
  BASE_API_URL: process.env.VITE_BASE_API_URL ?? '/api',
  ENVIRONMENT_NAME: process.env.VITE_ENVIRONMENT_NAME,
  SENTRY_DSN: process.env.VITE_SENTRY_DSN,
  WEB_APP_URL: process.env.VITE_WEB_APP_URL ?? '',
  EMAIL_ASSETS_URL: process.env.VITE_EMAIL_ASSETS_URL ?? '/email-assets',
  PUBLIC_URL: process.env.PUBLIC_URL,
  CONTENTFUL_SPACE: process.env.VITE_CONTENTFUL_SPACE,
  CONTENTFUL_ENV: process.env.VITE_CONTENTFUL_ENV,
  CONTENTFUL_TOKEN: process.env.VITE_CONTENTFUL_TOKEN,
  STRIPE_PUBLISHABLE_KEY: process.env.VITE_STRIPE_PUBLISHABLE_KEY ?? '',
  GOOGLE_ANALYTICS_TRACKING_ID: process.env.VITE_GOOGLE_ANALYTICS_TRACKING_ID,
};
