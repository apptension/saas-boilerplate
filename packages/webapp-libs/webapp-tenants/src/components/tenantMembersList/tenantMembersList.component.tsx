import { useQuery } from '@apollo/client';
import { Table, TableBody, TableHead, TableHeader, TableRow } from '@sb/webapp-core/components/ui/table';
import { FormattedMessage } from 'react-intl';

import { useCurrentTenant } from '../../providers';
import { MembershipEntry } from './membershipEntry';
import { tenantMembersListQuery } from './tenantMembersList.graphql';

export const TenantMembersList = () => {
  const { data: currentTenant } = useCurrentTenant();
  const { data, refetch } = useQuery(tenantMembersListQuery, {
    fetchPolicy: 'cache-and-network',
    variables: {
      id: currentTenant?.id ?? '',
    },
    skip: !currentTenant,
  });

  const memberships = data?.tenant?.userMemberships?.filter((membership) => !!membership) ?? [];
  return (
    <Table>
      <TableHeader>
        <TableRow>
          <TableHead>
            <FormattedMessage defaultMessage="Username" id="Tenant members list / Username" />
          </TableHead>
          <TableHead>
            <FormattedMessage defaultMessage="Role" id="Tenant members list / Role" />
          </TableHead>
          <TableHead>
            <FormattedMessage defaultMessage="Invitation accepted" id="Tenant members list / Invitation accepted" />
          </TableHead>
          <TableHead>
            <FormattedMessage defaultMessage="Actions" id="Tenant members list / Actions" />
          </TableHead>
        </TableRow>
      </TableHeader>
      <TableBody>
        {memberships.map(
          (membership) =>
            membership && <MembershipEntry membership={membership} key={membership.id} onAfterUpdate={refetch} />
        )}
      </TableBody>
    </Table>
  );
};
