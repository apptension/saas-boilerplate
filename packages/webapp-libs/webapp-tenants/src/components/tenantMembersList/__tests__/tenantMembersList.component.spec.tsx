import { TenantUserRole } from '@sb/webapp-api-client';
import { currentUserFactory, fillCommonQueryWithUser } from '@sb/webapp-api-client/tests/factories';
import { composeMockedQueryResult, makeId } from '@sb/webapp-api-client/tests/utils';
import { RoutesConfig } from '@sb/webapp-core/config/routes';

import { membershipFactory, tenantFactory } from '../../../tests/factories/tenant';
import { createMockRouterProps, render } from '../../../tests/utils/rendering';
import { TenantMembersList } from '../tenantMembersList.component';
import { tenantMembersListQuery } from '../tenantMembersList.graphql';

describe('TenantMembersList: Component', () => {
  const Component = () => <TenantMembersList />;

  it('should render list of memberships', async () => {
    const currentUserMembership = membershipFactory({ role: TenantUserRole.OWNER });
    const tenant = tenantFactory({ membership: currentUserMembership });
    const user = currentUserFactory({ tenants: [tenant] });
    const commonQueryMock = fillCommonQueryWithUser(user);
    const routerProps = createMockRouterProps(RoutesConfig.home, { tenantId: tenant.id });

    const listQueryVariables = {
      id: tenant.id,
    };

    const listQueryData = {
      tenant: {
        userMemberships: [
          membershipFactory({
            role: TenantUserRole.OWNER,
            inviteeEmailAddress: null,
            userId: user.id,
            firstName: user.firstName,
            lastName: user.lastName,
            userEmail: user.email,
            avatar: null,
          }),
          membershipFactory({
            role: TenantUserRole.ADMIN,
            inviteeEmailAddress: null,
            userId: makeId(32),
            firstName: 'Firstname 1',
            lastName: 'Firstname 1',
            userEmail: null,
            avatar: null,
          }),
          membershipFactory({
            role: TenantUserRole.MEMBER,
            invitationAccepted: false,
            inviteeEmailAddress: 'example@example.com',
            userId: makeId(32),
            firstName: null,
            lastName: null,
            userEmail: null,
            avatar: null,
          }),
        ],
      },
    };

    const requestListMock = composeMockedQueryResult(tenantMembersListQuery, {
      variables: listQueryVariables,
      data: listQueryData,
    });

    const { container, waitForApolloMocks } = render(<Component />, {
      apolloMocks: [commonQueryMock, requestListMock],
      routerProps,
    });
    await waitForApolloMocks();
    expect(container).toMatchSnapshot();
  });
});
