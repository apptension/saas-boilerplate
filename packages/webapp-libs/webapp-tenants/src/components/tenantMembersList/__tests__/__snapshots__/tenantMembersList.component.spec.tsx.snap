// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenantMembersList: Component should render list of memberships 1`] = `
<div>
  <div
    class="relative w-full overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
    >
      <thead
        class="[&_tr]:border-b"
      >
        <tr
          class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
        >
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Username
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Role
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Invitation accepted
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      >
        <tr
          class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
        >
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            testFirstName testLastName
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            Owner
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            <div
              class="flex items-center"
            >
              <svg
                class="lucide lucide-user-check mr-2 w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="9"
                  cy="7"
                  r="4"
                />
                <polyline
                  points="16 11 18 13 22 9"
                />
              </svg>
              Yes
            </div>
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none hover:bg-accent hover:text-accent-foreground h-8 w-8 p-0"
              data-state="closed"
              id="radix-:r0:"
              role="button"
              type="button"
            >
              <svg
                class="lucide lucide-grip-horizontal h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="5"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="12"
                  cy="15"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="15"
                  r="1"
                />
                <circle
                  cx="5"
                  cy="15"
                  r="1"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
        >
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            Firstname 1 Firstname 1
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            Admin
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            <div
              class="flex items-center"
            >
              <svg
                class="lucide lucide-user-check mr-2 w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="9"
                  cy="7"
                  r="4"
                />
                <polyline
                  points="16 11 18 13 22 9"
                />
              </svg>
              Yes
            </div>
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none hover:bg-accent hover:text-accent-foreground h-8 w-8 p-0"
              data-state="closed"
              id="radix-:r2:"
              role="button"
              type="button"
            >
              <svg
                class="lucide lucide-grip-horizontal h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="5"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="12"
                  cy="15"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="15"
                  r="1"
                />
                <circle
                  cx="5"
                  cy="15"
                  r="1"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
        >
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            example@example.com
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            Member
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            <div
              class="flex items-center"
            >
              <svg
                class="lucide lucide-hourglass mr-2 w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 22h14"
                />
                <path
                  d="M5 2h14"
                />
                <path
                  d="M17 22v-4.172a2 2 0 0 0-.586-1.414L12 12l-4.414 4.414A2 2 0 0 0 7 17.828V22"
                />
                <path
                  d="M7 2v4.172a2 2 0 0 0 .586 1.414L12 12l4.414-4.414A2 2 0 0 0 17 6.172V2"
                />
              </svg>
              No
            </div>
          </td>
          <td
            class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
          >
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none hover:bg-accent hover:text-accent-foreground h-8 w-8 p-0"
              data-state="closed"
              id="radix-:r4:"
              role="button"
              type="button"
            >
              <svg
                class="lucide lucide-grip-horizontal h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="5"
                  cy="9"
                  r="1"
                />
                <circle
                  cx="12"
                  cy="15"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="15"
                  r="1"
                />
                <circle
                  cx="5"
                  cy="15"
                  r="1"
                />
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    aria-label="Notifications (F8)"
    role="region"
    style="pointer-events: none;"
    tabindex="-1"
  >
    <ol
      class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
      data-testid="toaster"
      tabindex="-1"
    />
  </div>
</div>
`;
